### ISSUE TITLE
Component won't update

### ISSUE THREAD

**Issue by cslinmiso** [jcoreio/redux-form-normalize-on-blur#3]
I have this situation that I had to dynamically change my component, and that's when this component renders the wrong component until it gains focus.

https://codesandbox.io/s/q807n8nx64

Enter something in the Normalize field and click on ReadOnly switch, it should switch to ReadOnly component but it only changes when it gains focus.

**Comment by jedwards1211** [comment]
Thanks for the report and the sandbox!  It's definitely a bug, working on a fix right now

**Comment by cslinmiso** [comment]
My current solution is like this
```

const Field = (props) => {
  const { component, normalizeOnBlur } = props
  if (normalizeOnBlur) {
    return <EnhancedField {...props} normalizeOnBlur={normalizeOnBlur} component={enhanceNormalizeOnBlur} originalComponent={component} />
  }
  return <EnhancedField component={component} {...props} />
}
```

**Comment by jedwards1211** [comment]
I don't understand why you doubt it, it fixed tbe problem when I used the new version in your sandbox, and I added a test that mimics the problem you were having.

Redux-form's `Field` does rerender when its `component` prop changes, and the memoization ensures that the blur handler wrapper component changes anytime the component you passed in changes.

Your solution also works but the reason I didn't pass an extra prop like `originalComponent` is, if a dev happened to be using that prop for something else, it would break their code.
